api/agent/timer_desired_state_feedback_test.go

```diff
@@ -0,0 +1,322 @@
+		mockClient := mocks.NewMockUpdateAgentClient(mockCtr)
+		interval := time.Second
+		updAgent := &updateAgent{
+			client: mockClient,
+		}
+		expectedNotifier := &desiredStateFeedbackNotifier{
+			interval:        interval,
+			agent:           updAgent,
+			actions:         actions,
+			reportedActions: reportedActions,
+		}
+		actualNotifier := newDesiredStateFeedbackNotifier(interval, updAgent)
+
+		assert.Equal(t, expectedNotifier, actualNotifier)
+	})
+}
+
+func TestDesiredStateFeedbackTimerSet(t *testing.T) {
+	mockCtr := gomock.NewController(t)
+	defer mockCtr.Finish()
+
+	actions := []*types.Action{
+		{
+			Component: &types.Component{
+				ID:      "action 1 component",
+				Version: "1.0.0",
+			},
+		},
+	}
+
+	actions2 := []*types.Action{
+		{
+			Component: &types.Component{
+				ID:      "action2",
+				Version: "2.0.0",
+			},
+		},
+	}
+
+	reportedActions := []*types.Action{
+		{
+			Component: &types.Component{
+				ID:      "action 1 component",
+				Version: "1.0.0",
+			},
+		},
+	}
+	reportedActions2 := []*types.Action{
+		{
+			Component: &types.Component{
+				ID:      "action2",
+				Version: "2.0.0",
+			},
+		},
+	}
+
+	t.Run("test_set_internal_timer_not_nil", func(t *testing.T) {
+		mockClient := mocks.NewMockUpdateAgentClient(mockCtr)
+		interval := time.Second
+		updAgent := &updateAgent{
+			client: mockClient,
+		}
+
+		notifier := newDesiredStateFeedbackNotifier(interval, updAgent)
+		notifier.internalTimer = time.AfterFunc(interval, func() {})
+
+		notifier.set(testActivityID, actions)
+
+		assert.Equal(t, actions, notifier.actions)
+		if notifier.internalTimer != nil {
+			if !notifier.internalTimer.Stop() {
+				<-notifier.internalTimer.C
+			}
+		}
+	})
+
+	t.Run("test_set_internal_timer_nil", func(t *testing.T) {
+		mockClient := mocks.NewMockUpdateAgentClient(mockCtr)
+		interval := time.Second
+		updAgent := &updateAgent{
+			client: mockClient,
+		}
+
+		notifier := newDesiredStateFeedbackNotifier(interval, updAgent)
+		mockClient.EXPECT().SendDesiredStateFeedback(gomock.Any(), gomock.Any()).Times(1)
+		notifier.set(testActivityID, actions)
+
+		assert.Equal(t, reportedActions, notifier.reportedActions)
+		assert.Equal(t, testActivityID, notifier.activityID)
+		if notifier.internalTimer != nil {
+			if !notifier.internalTimer.Stop() {
+				<-notifier.internalTimer.C
+			}
+		}
+	})
+
+	t.Run("test_actions_change_during_timeout", func(t *testing.T) {
+		mockClient := mocks.NewMockUpdateAgentClient(mockCtr)
+		interval := 2 * time.Second
+		updAgent := &updateAgent{
+			client: mockClient,
+		}
+		notifier := newDesiredStateFeedbackNotifier(interval, updAgent)
+		mockClient.EXPECT().SendDesiredStateFeedback(gomock.Any(), gomock.Any()).Times(1)
+		notifier.set(testActivityID, actions)
+		assert.Equal(t, actions, notifier.actions)
+		assert.Equal(t, reportedActions, notifier.reportedActions)
+		assert.Equal(t, testActivityID, notifier.activityID)
+		time.Sleep(1 * time.Second)
```

Comment:
after the refactoring mentioned above, please use the interval constants in the `sleep` functions as well